### ISSUE TITLE
Close the window when ComboBox in ContentDialog being opened crashes application

### ISSUE THREAD

**Issue by HO-COOH** [microsoft/microsoft-ui-xaml#8605]
### Describe the bug

Put a `ComboBox` in a `ContentDialog`, and close the window if the `ComboBox` is currently opened will crash the app.

### Steps to reproduce the bug

1. Create a new WinUI3 C# project
2. In `MainWindow.xaml.cs`
```cs
    public sealed partial class MainWindow : Window
    {
        public MainWindow()
        {
            this.InitializeComponent();
        }

        private void myButton_Click(object sender, RoutedEventArgs e)
        {
            var comboBox = new ComboBox();
            comboBox.Items.Add("Item");
            var dialog = new ContentDialog()
            {
                Content = comboBox,
                XamlRoot = myButton.XamlRoot 
            };
            dialog.ShowAsync();
        }
    }
```
3. Click to open the combobox, and then close the window.
4. App crash

### Screenshots

![image](https://github.com/microsoft/microsoft-ui-xaml/assets/42881734/d66b4d31-861b-41d6-8ba0-c4cf8176fe8a)

### NuGet package version

WinUI 3 - Windows App SDK 1.4 Experimental 1: 1.4.230518007-experimental1

### Windows version

Windows 11 (22H2): Build 22621

### Additional context

An Issue I found might be similar:  https://github.com/microsoft/microsoft-ui-xaml/issues/6321

**Comment by HO-COOH** [comment]
A finicky workaround I found: Intercept the `Window.Closed` message, and hide the dialog first, this would bring some more exceptions raises internally, but it does not crash the app.
```
void WindowClose()
{
    m_dialog.Hide();
}
```

**Comment by HO-COOH** [comment]
This issue is so universal that affecting some major apps, like [Powertoys](https://learn.microsoft.com/en-us/windows/powertoys/) and [Files](https://github.com/files-community/Files).

**Comment by HO-COOH** [comment]
https://github.com/microsoft/microsoft-ui-xaml/assets/42881734/1f08a3c3-a8ca-40a5-ac19-7187d4f1ed82


https://github.com/microsoft/microsoft-ui-xaml/assets/42881734/a120b96c-c558-4258-b911-d923c5fd7960

**Comment by codendone** [comment]
The problem here is the popup tries to close itself when the popup root's children list is locked. XAML probably needs to defer closing the popup until a safe time.

**Comment by codendone** [comment]
This is fixed for 1.5.
